### PR TITLE
Added app_utils tests

### PR DESCRIPTION
**Pull request by cszsol**

### CODE REVIEW

File: swarm_copy/tools/bluenaas_memodel_getall.py [hunk 1]
```diff
@@ -29,9 +29,11 @@ class InputMEModelGetAll(BaseModel):
     page_size: int = Field(
         default=20, description="Number of results returned by the API."
     )
-    model_type: Literal["single-neuron-simulation", "synaptome-simulation"] = Field(
```
Comment (WonderPG) [inline swarm_copy/tools/bluenaas_memodel_getall.py:32]: As in other MRs, the name is not fitting.

File: swarm_copy_tests/app/test_dependencies.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    monkeypatch.setenv("NEUROAGENT_DB__PASSWORD", "Doe")
+    monkeypatch.setenv("NEUROAGENT_DB__HOST", "localhost")
+    monkeypatch.setenv("NEUROAGENT_DB__PORT", "5000")
+    monkeypatch.setenv("NEUROAGENT_DB__NAME", "test")
+
+    settings = Settings()
+    result = get_connection_string(settings)
+    assert (
+        result == "http://John:Doe@localhost:5000/test"
+    ), "must return fully formed connection string"
```
Comment (WonderPG) [inline swarm_copy_tests/app/test_dependencies.py:171]: Couple of tests are missing:
- `get_vlab_and_project`
- `get_starting_agent`
- `get_context_variables` (arguably no need to test it)
- `get_openai_client` (arguably no need to test it)
- `get_agents_routine` (arguably no need to test it)
- `get_kg_token`

File: swarm_copy_tests/app/test_main.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def test_readyz(app_client):
+    response = app_client.get(
+        "/",
+    )
+
+    body = response.json()
+    assert isinstance(body, dict)
+    assert body["status"] == "ok"
```
Comment (WonderPG) [inline swarm_copy_tests/app/test_main.py:17]: Lifespan should be tested as well

File: swarm_copy_tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+from httpx import AsyncClient
+from langchain_core.language_models.fake_chat_models import GenericFakeChatModel
+from langchain_core.messages import AIMessage
+from sqlalchemy import MetaData
+from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
+
+from neuroagent.app.config import Settings
+from neuroagent.app.dependencies import get_kg_token, get_settings
+from neuroagent.app.main import app
+from neuroagent.tools import GetMorphoTool
```
Comment (WonderPG) [inline swarm_copy_tests/conftest.py:18]: Nothing should be imported from neuroagent.

File: swarm_copy_tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+import json
+from pathlib import Path
+
+import pytest
+import pytest_asyncio
+from fastapi.testclient import TestClient
+from httpx import AsyncClient
+from langchain_core.language_models.fake_chat_models import GenericFakeChatModel
+from langchain_core.messages import AIMessage
```
Comment (WonderPG) [inline swarm_copy_tests/conftest.py:11]: Nothing should be imported from langchain.

File: swarm_copy_tests/conftest.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+            "url": "http://fake_url",
+            "search_size": 2,
+            "httpx_client": async_client,
+            "token": "fake_token",
+            "brainregion_path": brain_region_json_path,
+        }
+    )
+
+    yield FakeFuntionChatModel(messages=iter(fake_responses)), [tool], fake_responses
+    await async_client.aclose()
```
Comment (WonderPG) [inline swarm_copy_tests/conftest.py:210]: Not used anymore as this is langchain's logic.

**Review by WonderPG — COMMENTED** [review]
Couple of comments, thank you !

File: swarm_copy/tools/bluenaas_memodel_getall.py (hunk 1 above)
Comment (kanesoban) [reply swarm_copy/tools/bluenaas_memodel_getall.py:32]: I already fixed this in the other MR:
https://github.com/BlueBrain/neuroagent/pull/44
When that PR is merged I will merge that onto this PR.

File: swarm_copy_tests/app/test_dependencies.py (hunk 2 above)
Comment (kanesoban) [reply swarm_copy_tests/app/test_dependencies.py:175]: I added tests for 

- get_vlab_and_project
- get_starting_agent
- get_kg_token

For the other 3, I don't think there is much point.

File: swarm_copy/app/dependencies.py [hunk 7]
```diff
@@ -195,6 +195,7 @@ async def get_vlab_and_project(
         }
     else:
         thread_id = request.path_params.get("thread_id")
+        logging.error(f"*** Thread id: {thread_id}, User id: {user_id} ***")
```
[the remark this answers is not in the document]
Comment (kanesoban) [reply swarm_copy/app/dependencies.py:198]: This is a debugging message that I left here by accident. Good that it was noticed, I removed it.

**Review by WonderPG — APPROVED** [review]

**Review by kanesoban — APPROVED** [review]

**Pull request merged**